src/components/Table.js

```diff
@@ -0,0 +1,202 @@
+import classnames from 'classnames';
+import { useEffect, useRef } from 'preact/hooks';
+
+import { refElement } from '../util/typing';
+
+import { Scrollbox } from './containers';
+import { Spinner } from './Spinner';
+
+/**
+ * @typedef TableColumn
+ * @prop {string} label - Header label for the column
+ * @prop {string} [classes] - Additional CSS classes for the column's `<th>` element
+ */
+
+/**
+ * @template Item
+ * @typedef TableProps
+ * @prop {string} accessibleLabel - An accessible label for the table
+ * @prop {string} [classes] - Extra CSS classes to apply to the outermost
+ *   element, which is a <Scrollbox> div
+ * @prop {TableColumn[]} columns - The columns to display in this table
+ * @prop {boolean} [isLoading] - Show an indicator that data for the table is
+ *   currently being fetched
+ * @prop {Item[]} items -
+ *   The items to display in this table, one per row. `renderItem` defines how
+ *   information from each item is represented as a series of table cells.
+ * @prop {(it: Item, selected: boolean) => any} renderItem -
+ *   A function called to render each item as the contents of a table row.
+ *   The result should be a list of `<td>` elements (one per column) wrapped inside a Fragment.
+ * @prop {Item|null} selectedItem - The currently selected item from `items`
+ * @prop {(it: Item) => void} onSelectItem -
+ *   Callback invoked when the user changes the selected item
+ * @prop {(it: Item) => void} onUseItem -
+ *   Callback invoked when a user chooses to use an item by double-clicking it
+ *   or pressing Enter while it is selected
+ * @prop {string} [tableClasses] - Extra CSS classes to apply to the <table>
```

Comment:
I struggled with this. The API I've been trying to establish is that a component's `classes` prop will set CSS classes on the outermost element in the rendered component. My goal was to make it very clear to the author what element the `classes` will apply to whenever they're using a component from this library of components—the outermost component element.  This doesn't map well to components that have a wrapper element, as you see here. The main "thing" in this component is certainly the `table` element, but that breaks the expectation that `classes` will always get applied to the outermost element. 

OTOH, it may be that an author would _expect_ `classes` to apply to the `table`, and, if that didn't do what they needed, they could look at the implementation and see that there is a `containerClasses` prop... 🤔 as you've suggested here.

I think your suggestion probably makes more sense. Perhaps what I should aim for is:

* `classes` will apply classes to the element that corresponds to the component (e.g. `Table`) or to the outermost container element rendered by the component
* `containerClasses` (obviously not implemented by all components) can be used to apply CSS classes to the outermost component container element in cases where it's needed.

Reply:
For me your last suggestion makes more sense, however, it maybe better to be consistent and if `classes` have been always applied to the outermost element maybe is better to leave it like that.

Reply:
I'm updating it to take `classes` for the `<table>` element and `containerClasses` for the wrapping element.